app/controllers/split_checkout_controller.rb

```diff
@@ -177,10 +189,35 @@ def shipping_method_ship_address_not_required?
     selected_shipping_method.first.require_ship_address == false
   end
 
+  def add_voucher
+    return unless payment_step? && params[:order] && params[:order][:voucher_code]
+
+    # Fetch Voucher
+    voucher = Voucher.find_by(code: params[:order][:voucher_code], enterprise: @order.distributor)
+
+    if voucher.nil?
+      @order.errors.add(:voucher, I18n.t('split_checkout.errors.voucher_not_found'))
```

Comment:
I did not know that, TIL!  However, because I renamed the `voucher` association I can't used your solution. Rails complains about missing `voucher` method, and using `voucher_adjustments` would be confusing for customer. I left it as it is.

Reply:
> using voucher_adjustments would be confusing for customer.

Rails got you covered. You can define the displayed label of an attribute in en.yml. Examples:
https://github.com/openfoodfoundation/openfoodnetwork/blob/e91e827f29d05bcd4754f3ca1388a82828ab637a/config/locales/en.yml#L24-L31

But this PR is big enough. No need to change that now.

Reply:
Interesting, I feel like the more I learn the more I realize I don't know that much about rails ...